In [2]:
import numpy as np
from scipy import stats

## Задача 5-1.
Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.Найти доверительный интервал для оценки математического ожидания с надежностью 0.95,если выборочная средняя M = 80, а объем выборки n = 256.

### Решение:
Найдем доверительный интервал согласно формуле:$$P \left(\overline{X} + Z_{\dfrac \alpha 2} \cdot \dfrac \sigma {\sqrt{n}} 
\leq M(X) \leq \overline{X} + Z_{\dfrac \alpha 2} \cdot \dfrac \sigma {\sqrt{n}}\right)$$

где $\overline{X} = 80$, $S = 16$, $n = 256$, а $Z$ найдем по таблице значений: $Z = 1,96$

In [3]:
std_ = 16
mean_ = 80
n = 256
print ('Доверительный интервал равен:', (mean_ - 1.96*std_/np.sqrt(n), mean_ + 1.96*std_/np.sqrt(n)))

Доверительный интервал равен: (78.04, 81.96)


#### Ответ 5-1: Доверительный интервал равен: (78.04, 81.96)

## Задача 5-2.
В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

### Решение:
[1] Найдем доверительный интервал согласно формуле:
$M \pm t_{\frac \alpha 2} \cdot \dfrac \sigma {\sqrt{n}}$, где:
- $n = 10$
- $1-\alpha = 95\%$
- $\alpha = 5\%$

In [13]:
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
n = 10
m = x.mean()
print(f'Математическое ожидание M(X) = {m}')
sigma = np.sqrt(((x - m)**2).sum()/(n-1))
print (f'Среднее квадратичное отклонение 𝜎 = {sigma}')

Математическое ожидание M(X) = 6.590000000000001
Среднее квадратичное отклонение 𝜎 = 0.4508017549014448


[2] Табличное значение коэффициента Стьюдента:

In [17]:
t = 2.262
print (f'Коэффициента Стьюдента t = {t}')

Коэффициента Стьюдента t = 2.262


In [18]:
print ('Доверительный интервал равен:', (m - (t*sigma/np.sqrt(n)), m + (t*sigma/np.sqrt(n))))

Доверительный интервал равен: (6.267538255912426, 6.912461744087575)


#### Ответ 5-2: Доверительный интервал, построенный с учётом доверительной вероятности 95%, равен (6.2675, 6.91246). Истинное значение величины лежит пределах этого доверительного интервала.

## 3 и 4 задачи решать через тестирование гипотезы

## Задача 5-3.
Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр $17~мм$. Используя односторонний критерий с $\alpha = 0,05$, проверить эту гипотезу, если в выборке из $n=100$ шариков средний диаметр оказался равным $17,5~мм$, а дисперсия известна и равна $4~кв.мм$.

### Решение:
[1] Сформулируем нулевую и альтернативную гипотезы:
- Гипотеза $H_0$: средний диаметр шарика для подшипников равен 17 мм.

- Гипотеза $H_1$: средний диаметр шарика для подшипников больше 17 мм.

[2] Зададим некоторую функцию Т от выборки:
Предполагается, что количество шариков большое, то распределение диаметра шариков является нормальным или близким к нему. Т.к. случайная величина диаметра имеет нормальное распределение, то среднее значение повыборке $T$ будет также иметь нормальное распределение с параметрами $a = 17$ и $\sigma = \sqrt{\dfrac{D(X)}{n}} = \sqrt {\dfrac{4}{100}} = 0.2 $

In [4]:
a = 17
m = 17.5
n = 100
D = 4
sigma = np.sqrt(D/n)
print(f'Среднее квадратичное отклонение 𝜎 = {sigma}')

Среднее квадратичное отклонение 𝜎 = 0.2


[3] Фиксируем уровень значимости $\alpha$:
- допустимая для данной задачи вероятность ошибки первого рода (чаще всего 0.01, 0.05 или 0.1):

$\alpha = 0,05$

[4] Определяеем критическую область $\Omega_\alpha = (t_{1 - \alpha}, \infty)$.
Выбранный уровень значимости $\alpha = 0,025$ означает, что для проверки гипотезы о среднем значении a = 17 требуется, чтобы с вероятностью 95% выборочное среднее попадало в правосторонний интервал, который (по правилу 2-х сигм) равен $(a - 2 \cdot \sigma,  a + 2 \cdot \sigma)$.

То есть критическая область лежит правее точки $a + 2 \cdot \sigma $ и если выборочное среднее = 17,5 попадёт в критическую область, то основная гипотеза отвергается.

In [5]:
print(f'Критическая область  Ω𝛼 = {a - 2 * sigma, a + 2 * sigma}')

Критическая область  Ω𝛼 = (16.6, 17.4)


[5] Проводем статистический тест:
для конкретной выборки $X$ считаем значение $T(X)$ , и если оно принадлежит $\Omega_\alpha$, то заключаем, что данные противоречат гипотезе $H_0$, и принимается гипотеза $H_1$.

In [6]:
stats.norm.ppf(1 - 0.025, 17, 0.2)
print(f'Статистический тест: {stats.norm.ppf(1 - 0.025, 17, 0.2)}')

Статистический тест: 17.39199279690801


#### Ответ 5-3: Гипотеза о том, что шарики для подшипников имеют средний диаметр 17 мм, при выбранном уровне значимости $\alpha = 0,025$ отвергается.

## Задача 5-4.
Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек.
Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

### Решение:
[1] Найдем доверительный интервал, чтобы проверить гипотезу, что среднее значение веса пачки 200 г:

In [7]:
sample = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
m = sample.mean()
n = 10
sigma = np.sqrt(((sample - m)**2).sum()/(n-1))
print ('Среднее квадратичное отклонение 𝜎 = ', sigma)

Среднее квадратичное отклонение 𝜎 =  4.453463071962462


[2] Табличное значение коэффициента Стьюдента:

In [8]:
t_int = 2.8214

[3] Проверим, что коэффицент Стьюдента = 2.8214, соответствует вероятности:

In [9]:
t = stats.t(n-1)
t.cdf(2.8214)

0.9899993808040004

In [10]:
print ('Доверительный интервал равен:', (m - t_int*sigma/np.sqrt(n), m + t_int*sigma/np.sqrt(n)))

Доверительный интервал равен: (194.5265978950862, 202.4734021049138)


#### Ответ 5-4: Доверительный интервал, построенный с учётом доверительной вероятности 99%, равен (194.5266, 202.4734), включает значение предполагаемого среднего значения а = 200 г, следовательно утверждение продавца, что средний вес пачки составляет 200 г, верно.